static/app/components/confirm.tsx

```diff
@@ -226,66 +226,57 @@ type ModalProps = ModalRenderProps &
     | 'onRender'
   >;
 
-type ModalState = {
-  /**
-   * The callback registered from the rendered message to call
-   */
-  confirmCallback: null | (() => void);
-  /**
-   * Is confirm button disabled
-   */
-  disableConfirmButton: boolean;
-};
-
-class ConfirmModal extends Component<ModalProps, ModalState> {
-  state: ModalState = {
-    disableConfirmButton: !!this.props.disableConfirmButton,
-    confirmCallback: null,
-  };
-
-  componentDidMount() {
-    this.props.onRender?.();
-  }
-
-  confirming = false;
-
-  handleClose = () => {
-    const {disableConfirmButton, onCancel, closeModal} = this.props;
-
+function ConfirmModal({
+  Header,
+  Body,
+  Footer,
+  priority,
+  confirmText,
+  cancelText,
+  header,
+  isDangerous,
+  renderConfirmButton,
+  renderCancelButton,
+  disableConfirmButton,
+  onCancel,
+  onConfirm,
+  renderMessage,
+  message,
+  closeModal,
+}: ModalProps) {
+  const confirmCallbackRef = useRef<() => void>(() => {});
+  const isConfirmingRef = useRef(false);
+  const [shouldDisableConfirmButton, setShouldDisableConfirmButton] =
+    useState(disableConfirmButton);
```

Comment:
This state seems unnecessary, but `renderMessage` can call `disableConfirmButton` which makes things more complicated. Another think I think should be removed but not going to do here